server/websocket.js

```diff
@@ -61,98 +24,202 @@ const setupWebSocketServer = server => {
-            ws.send(
-              JSON.stringify({ type: 'error', error: 'Unsupported platform' })
+          // Use proxy URL from the message, fallback to default
+          const effectiveProxyUrl = data.proxyUrl || PROXY_URL;
+          console.info('Using proxy URL:', effectiveProxyUrl);
+
+          try {
+            console.info(
+              `Connecting to proxy streaming endpoint at ${effectiveProxyUrl}`
             );
-            return;
-          }
 
-          // eslint-disable-next-line no-console
-          console.info('Spawning process with command', command);
-          const process = spawn(
-            command.split(' ')[0],
-            command.split(' ').slice(1),
-            { shell: true }
-          );
-
-          // Store the session
-          captureSessions.set(sessionId, { process, ws });
-          // eslint-disable-next-line no-console
-          console.info('Session stored', Array.from(captureSessions.keys()));
-
-          // Handle process output
-          process.stdout.on('data', data => {
-            const output = data.toString();
-
-            // eslint-disable-next-line no-console
-            console.info('stdout', output);
-            ws.send(JSON.stringify({ type: 'utterance', data: output }));
-          });
-
-          process.stderr.on('data', data => {
-            console.error('process.stderr', data.toString());
-          });
-
-          process.on('error', error => {
-            console.error('process.error', error);
-            ws.send(JSON.stringify({ type: 'error', error: error.message }));
-            captureSessions.delete(sessionId);
-          });
+            // Create abort controller for this session
+            const abortController = new AbortController();
+
+            // Store the session
+            captureSessions.set(sessionId, { abortController, ws });
+            console.info('Session stored', Array.from(captureSessions.keys()));
+
+            // Start streaming from proxy using native HTTP (axios closes SSE streams)
+            const proxyUrl = new URL(
+              `${effectiveProxyUrl}/stream-capture-utterances`
+            );
+            const postData = JSON.stringify({});
+
+            const options = {
+              hostname: proxyUrl.hostname,
+              port:
+                proxyUrl.port || (proxyUrl.protocol === 'https:' ? 443 : 80),
+              path: proxyUrl.pathname,
+              method: 'POST',
+              headers: {
+                'Content-Type': 'application/json',
+                'Content-Length': Buffer.byteLength(postData),
+                Accept: 'text/event-stream',
+                'Cache-Control': 'no-cache'
+              }
+            };
+
+            // Use https for ngrok or similar URLs, http for localhost
+            const requestModule = proxyUrl.protocol === 'https:' ? https : http;
+
+            const req = requestModule.request(options, response => {
+              console.info(
```

Comment:
We probably want to make a lot of logging added in this PR configurable. This could be a follow up

Reply:
Good thought!